src/clocksi_vnode.erl

```diff
@@ -581,15 +582,15 @@ update_materializer(DownstreamOps, Transaction, TxCommitTime) ->
     end.
 
 %% Internal functions
-filter_updates_per_key(Updates, Key) ->
-    FilterMapFun = fun({KeyPrime, _Type, Op}) ->
-        case KeyPrime == Key of
-            true -> {true, Op};
-            false -> false
-        end
-    end,
-    lists:filtermap(FilterMapFun, Updates).
-
+reverse_and_filter_updates_per_key(Updates, Key) ->
+    lists:foldl(fun({KeyPrime, _Type, Op}, Acc) ->
+			case KeyPrime == Key of
+			    true ->
+				[Op | Acc];
+			    false ->
+				Acc
+			end
+		end, [], Updates).
 
```

Comment:
Why did you rewrite this function? The new version is equivalent, it only has a different name.